src/compute-types/src/dataflows.rs

```diff
@@ -67,19 +67,21 @@ pub struct DataflowDescription<P, S: 'static = (), T = mz_repr::Timestamp> {
 impl<T> DataflowDescription<Plan<T>, (), mz_repr::Timestamp> {
     /// Tests if the dataflow refers to a single timestamp, namely
     /// that `as_of` has a single coordinate and that the `until`
-    /// value corresponds to the `as_of` value plus one.
+    /// value corresponds to the `as_of` value plus one, or `as_of`
+    /// is the maximum timestamp and is thus single.
     pub fn is_single_time(&self) -> bool {
         // TODO: this would be much easier to check if `until` was a strict lower bound,
         // and we would be testing that `until == as_of`.
         let Some(as_of) = self.as_of.as_ref() else {
             return false;
         };
-        !as_of.is_empty()
-            && as_of
-                .as_option()
-                .and_then(|as_of| as_of.checked_add(1))
-                .as_ref()
-                == self.until.as_option()
+        let Some(as_of) = as_of.as_option() else {
+            return false;
+        };
+        if as_of == &mz_repr::Timestamp::MAX {
+            return true;
```

Comment:
I think in this case we are also assuming that `self.until` is also the empty frontier. Can you add a `soft_assert_or_log!` entry for that condition before the `return true`?